docs/new_horizen/02-mainchain/05-on_chain_logic.md

```diff
@@ -0,0 +1,33 @@
+---
+title: New Horizen Mainchain On-chain Logic
+---
+
+# New Horizen Mainchain On-chain Logic
+
+## User flow
+The flow will be the following:
+
+1. A user (rollup / zkApp) submits the proof via our API.
+2. If the proof is valid, it is relayed by the consensus and eventually included in a Mainchain block; otherwise the transaction reverts with an error.
+3. The failing transaction will be included in the block anyway, and the user will pay fees for it (preventing DoS attacks).
```

Comment:
```suggestion
3. The failing transaction will be included in the block anyway, and the user will pay fees for it. This is to prevent DoS attacks.
```